### DISCUSSION TITLE
Poll drive directory for changes & regen w/o changing directories?

### DISCUSSION THREAD

**Discussion by benjamink** [tjboldt/Apple2-IO-RPi#124] [General]
The live update seems to require changing out of the directory for the live image & then changing back into it for the refresh/update to happen. Is this correct? If so, could logic be added to watch the directory for file changes & regenerate without the need to change out of & into the directory to refresh?

I'm thinking something along the lines of inotify but more native in the Go code.

(Moving from [ProDOS-Utilities discussion](https://github.com/tjboldt/ProDOS-Utilities/discussions/29))

**Comment by benjamink** [comment]
Related to changes in https://github.com/tjboldt/Apple2-IO-RPi/pull/122

**Comment by tjboldt** [comment]
Did you rebuild the apple2driver after pulling the latest from upstream? It sounds like it's still using the old code. You can verify this by `-RPI.COMMAND` and see if you get 0027 or 0028.

**Comment by benjamink** [comment]
I thought I had but I did get 0027 back so I must have missed that.  I went through the update steps in https://github.com/tjboldt/Apple2-IO-RPi/discussions/65 & am now seeing 0028.  But I'm not seeing new files show up in `CATALOG` output when I create/rename files in the `driveimage` directory.

I also noticed that after you symlinked the `.bas` files to the `apple2driver` directory that you are now dumped into that `apple2driver` directory when you use `-SHELL` instead of the `driveimage` directory.  Seems like ending up in the `driveimage` directory is preferable.

**Comment by tjboldt** [comment]
The .bas files are symlinked into the the driveimage directory... Not sure why you are seeing them in apple2driver...

https://github.com/tjboldt/Apple2-IO-RPi/tree/main/RaspberryPi/driveimage

Also, while not automatically polling yet, you can manually trigger a drive regen from the Apple II...

`]-RPI.COMMAND`
`]RPI a2drive 1 regen`

**Reply by benjamink** [reply]
As I re-read your response it dawned on me that the BASIC files being symlinked have nothing to do with where `-SHELL` puts me on the RPi.  Still not sure what would have changed so that `-SHELL` puts me in the `apple2driver` directory by default now, though.  

As for the `a2drive` command -makes sense.  I didn't think the functionality to auto-update was there yet like I was describing above.

**Comment by tjboldt** [comment]
Because it no longer uses the current working directory to generate, the code that set the initial working directory to `driveimage` was removed. It could be reinstated easily. I think for most people that makes sense although for me, I had `apple2driver` by default for easier in-system development.

**Reply by benjamink** [reply]
Maybe a config file in the future?

**Reply by tjboldt** [reply]
I updated it to default to the `driveimage` directory.

**Reply by benjamink** [reply]
Nice!  Updated & confirmed.

**Comment by benjamink** [comment]
I've noticed something else that I think crept in recently.  It seems the issue where it's impossible to do anything after exiting from `-SHELL` back to BASIC (https://github.com/tjboldt/Apple2-IO-RPi/pull/58 ).  It seems like access to the virtual disk actually stops working as entering Bitsy Bye after `-SHELL` also doesn't show any disk contents & hangs.  

Resetting the Apple2 gets things back to working again.

**Reply by benjamink** [reply]
Confirmed versions:

Firmware: 000F
RPI Command: 000E
Driver: 0028

**Comment by tjboldt** [comment]
This is definitely a bug. I doesn't happen every time but it is reproducible. It is different than bug #57 which would completely hang the RPi code. At least now a reboot of the Apple II fixes it due to commit #58. Would you care to open a new bug? If not, I can open one later.

**Comment by benjamink** [comment]
I can open one later or tomorrow but I'd be happy to. I'm hoping I can spend some time tomorrow troubleshooting it myself too.

**Comment by benjamink** [comment]
Opened bug issue: https://github.com/tjboldt/Apple2-IO-RPi/issues/130
